diff -ur busybox-1.7.4.orig/applets/applets.c busybox-1.7.4/applets/applets.c
--- busybox-1.7.4.orig/applets/applets.c	2007-09-16 13:48:10.000000000 -0500
+++ busybox-1.7.4/applets/applets.c	2008-03-11 13:53:18.000000000 -0500
@@ -23,7 +23,6 @@
 #warning Note that glibc is unsuitable for static linking anyway.
 #warning If you still want to do it, remove -Wl,--gc-sections
 #warning from top-level Makefile and remove this warning.
-#error Aborting compilation.
 #endif
 
 
diff -ur busybox-1.7.4.orig/Makefile busybox-1.7.4/Makefile
--- busybox-1.7.4.orig/Makefile	2007-11-23 22:34:41.000000000 -0600
+++ busybox-1.7.4/Makefile	2008-03-12 06:35:25.000000000 -0500
@@ -570,7 +570,7 @@
 quiet_cmd_busybox__ ?= LINK    $@
       cmd_busybox__ ?= $(srctree)/scripts/trylink $(CC) $(LDFLAGS) \
       -o $@ -Wl,-Map -Wl,$@.map \
-      -Wl,--warn-common -Wl,--sort-common -Wl,--gc-sections \
+      -Wl,--warn-common -Wl,--sort-common  \
       -Wl,--start-group $(busybox-all) -Wl,--end-group \
       $(LDLIBS)
 
