diff -ur busybox-1.00-rt-mdstart.plasmaroo/coreutils/Config.in busybox-1.00-rt-mdstart.plasmaroo.new/coreutils/Config.in
--- busybox-1.00-rt-mdstart.plasmaroo/coreutils/Config.in	2004-08-11 03:45:47.000000000 +0100
+++ busybox-1.00-rt-mdstart.plasmaroo.new/coreutils/Config.in	2006-01-11 23:35:23.000000000 +0000
@@ -469,7 +469,7 @@
 	  touch is used to create or change the access and/or
 	  modification timestamp of specified files.
 
-config CONFIG_TR
+config CONFIG_TR_X
 	bool "tr"
 	default n
 	help
diff -ur busybox-1.00-rt-mdstart.plasmaroo/coreutils/Makefile.in busybox-1.00-rt-mdstart.plasmaroo.new/coreutils/Makefile.in
--- busybox-1.00-rt-mdstart.plasmaroo/coreutils/Makefile.in	2004-10-08 08:45:14.000000000 +0100
+++ busybox-1.00-rt-mdstart.plasmaroo.new/coreutils/Makefile.in	2006-01-11 23:35:32.000000000 +0000
@@ -74,7 +74,7 @@
 COREUTILS-$(CONFIG_TEE)     	+= tee.o
 COREUTILS-$(CONFIG_TEST)    	+= test.o
 COREUTILS-$(CONFIG_TOUCH)   	+= touch.o
-COREUTILS-$(CONFIG_TR)      	+= tr.o
+COREUTILS-$(CONFIG_TR_X)      	+= tr.o
 COREUTILS-$(CONFIG_TRUE)    	+= true.o
 COREUTILS-$(CONFIG_TTY)     	+= tty.o
 COREUTILS-$(CONFIG_UNAME)   	+= uname.o
diff -ur busybox-1.00-rt-mdstart.plasmaroo/include/applets.h busybox-1.00-rt-mdstart.plasmaroo.new/include/applets.h
--- busybox-1.00-rt-mdstart.plasmaroo/include/applets.h	2006-01-11 23:29:22.000000000 +0000
+++ busybox-1.00-rt-mdstart.plasmaroo.new/include/applets.h	2006-01-11 23:35:30.000000000 +0000
@@ -586,7 +586,7 @@
 #ifdef CONFIG_TOUCH
 	APPLET(touch, touch_main, _BB_DIR_BIN, _BB_SUID_NEVER)
 #endif
-#ifdef CONFIG_TR
+#ifdef CONFIG_TR_X
 	APPLET(tr, tr_main, _BB_DIR_USR_BIN, _BB_SUID_NEVER)
 #endif
 #ifdef CONFIG_TRACEROUTE
