diff -ruN busybox-1.00-pre8/include/applets.h busybox-1.00-pre8-alpha/include/applets.h
--- busybox-1.00-pre8/include/applets.h	2004-01-27 04:22:20.000000000 -0500
+++ busybox-1.00-pre8-alpha/include/applets.h	2004-02-25 15:20:25.443610456 -0500
@@ -557,7 +557,7 @@
 	APPLET(sync, sync_main, _BB_DIR_BIN, _BB_SUID_NEVER)
 #endif
 #ifdef CONFIG_SYSCTL
-	APPLET(sysctl, sysctl_main, _BB_DIR_SBIN, _BB_SUID_NEVER)
+	//APPLET(sysctl, sysctl_main, _BB_DIR_SBIN, _BB_SUID_NEVER)
 #endif
 #ifdef CONFIG_SYSLOGD
 	APPLET(syslogd, syslogd_main, _BB_DIR_SBIN, _BB_SUID_NEVER)
