Index: gen_compile.sh
===================================================================
--- gen_compile.sh	(revision 528)
+++ gen_compile.sh	(working copy)
@@ -560,38 +560,38 @@
 	fi
 }
 
-compile_suspend() {
-	[ -f "${SUSPEND_BINCACHE}" ] && return
-	[ -f "${SUSPEND_SRCTAR}" ] ||
-		gen_die "Could not find SUSPEND source tarball: ${SUSPEND_SRCTAR}! Please place it there, or place another version, changing /etc/genkernel.conf as necessary!"
-	cd ${TEMP}
-	rm -rf ${SUSPEND_DIR} > /dev/null
-	/bin/tar -zxpf ${SUSPEND_SRCTAR} ||
-		gen_die 'Could not extract SUSPEND source tarball!'
-	[ -d "${SUSPEND_DIR}" ] ||
-		gen_die "SUSPEND directory ${DMRAID_DIR} is invalid!"
+#compile_suspend() {
+#	[ -f "${SUSPEND_BINCACHE}" ] && return
+#	[ -f "${SUSPEND_SRCTAR}" ] ||
+#		gen_die "Could not find SUSPEND source tarball: ${SUSPEND_SRCTAR}! Please place it there, or place another version, changing /etc/genkernel.conf as necessary!"
+#	cd ${TEMP}
+#	rm -rf ${SUSPEND_DIR} > /dev/null
+#	/bin/tar -zxpf ${SUSPEND_SRCTAR} ||
+#		gen_die 'Could not extract SUSPEND source tarball!'
+#	[ -d "${SUSPEND_DIR}" ] ||
+#		gen_die "SUSPEND directory ${DMRAID_DIR} is invalid!"
 
-	cd "${SUSPEND_DIR}"
-	if [ -f "${GK_SHARE}/pkg/suspend-0.5-Makefile.patch" ]
-	then
-		patch -p1 -i \
-			${GK_SHARE}/pkg/suspend-0.5-Makefile.patch \
-			|| gen_die "Failed patching suspend"
-	fi
+#	cd "${SUSPEND_DIR}"
+#	if [ -f "${GK_SHARE}/pkg/suspend-0.5-Makefile.patch" ]
+#	then
+#		patch -p1 -i \
+#			${GK_SHARE}/pkg/suspend-0.5-Makefile.patch \
+#			|| gen_die "Failed patching suspend"
+#	fi
 
-	print_info 1 'suspend: >> Compiling...'
-	compile_generic '' utils CC_FLAGS= LD_FLAGS=
+#	print_info 1 'suspend: >> Compiling...'
+#	compile_generic '' utils CC_FLAGS= LD_FLAGS=
 
-	print_info 1 '         >> Copying to bincache...'
-	mkdir -p "${TEMP}/bincache/sbin"
-	cp -f resume "${TEMP}/bincache/sbin" ||
-		gen_die 'Could not copy resume binary'
-	cd "${TEMP}/bincache"
-	/bin/tar -cjf "${SUSPEND_BINCACHE}" * ||
-		gen_die 'Could not create suspend binary cache'
-	cd "${TEMP}"
-	rm -rf bincache suspend-0.5
-}
+#	print_info 1 '         >> Copying to bincache...'
+#	mkdir -p "${TEMP}/bincache/sbin"
+#	cp -f resume "${TEMP}/bincache/sbin" ||
+#		gen_die 'Could not copy resume binary'
+#	cd "${TEMP}/bincache"
+#	/bin/tar -cjf "${SUSPEND_BINCACHE}" * ||
+#		gen_die 'Could not create suspend binary cache'
+#	cd "${TEMP}"
+#	rm -rf bincache suspend-0.5
+#}
 
 compile_devfsd() {
 	# I've disabled dietlibc support for the time being since the
Index: gen_configkernel.sh
===================================================================
--- gen_configkernel.sh	(revision 528)
+++ gen_configkernel.sh	(working copy)
@@ -122,9 +122,9 @@
 	fi
 
 	# This check isn't complete: SOFTWARE_SUSPEND has extra deps on some systems such as CPU hotplug
-	if isTrue ${CMD_SUSPEND}
-	then
-		sed -i ${KERNEL_DIR}/.config -e 's/#\? \?CONFIG_SOFTWARE_SUSPEND is.*/CONFIG_SOFTWARE_SUSPEND=y/g'
-	fi
+#	if isTrue ${CMD_SUSPEND}
+#	then
+#		sed -i ${KERNEL_DIR}/.config -e 's/#\? \?CONFIG_SOFTWARE_SUSPEND is.*/CONFIG_SOFTWARE_SUSPEND=y/g'
+#	fi
 
 }
Index: gen_cmdline.sh
===================================================================
--- gen_cmdline.sh	(revision 528)
+++ gen_cmdline.sh	(working copy)
@@ -79,7 +79,7 @@
   echo "	--lvm2			Include LVM2 support"
 #  echo "	--unionfs		Include UNIONFS support"
   echo "	--dmraid		Include DMRAID support"
-  echo "	--suspend		Include userspace suspend/resume (uswsusp) support"
+#  echo "	--suspend		Include userspace suspend/resume (uswsusp) support"
   echo "	--slowusb		Enables extra pauses for slow USB CD boots"
   echo "	--bootloader=grub	Add new kernel to GRUB configuration"
   echo "	--linuxrc=<file>	Specifies a user created linuxrc"
@@ -231,17 +231,17 @@
 		      CMD_LVM2=1
 		      print_info 2 "CMD_LVM2: $CMD_LVM2"
 	      ;;
-	      --suspend)
-		      if [ ! -e /etc/suspend.conf ]
-		      then
-			echo 'Error: --suspend requires sys-power/suspend to be installed'
-			echo '       on the host system; try "emerge sys-power/suspend".'
-			echo '       Once installed, configure the resume settings in /etc/suspend.conf'
-			exit 1
-		      fi
-	      	      CMD_SUSPEND=1
-		      print_info 2 "CMD_SUSPEND: $CMD_SUSPEND"
-	      ;;
+#	      --suspend)
+#		      if [ ! -e /etc/suspend.conf ]
+#		      then
+#			echo 'Error: --suspend requires sys-power/suspend to be installed'
+#			echo '       on the host system; try "emerge sys-power/suspend".'
+#			echo '       Once installed, configure the resume settings in /etc/suspend.conf'
+#			exit 1
+#		      fi
+#	      	      CMD_SUSPEND=1
+#		      print_info 2 "CMD_SUSPEND: $CMD_SUSPEND"
+#	      ;;
 	      --no-busybox)
 		      CMD_NO_BUSYBOX=1
 		      print_info 2 "CMD_NO_BUSYBOX: $CMD_NO_BUSYBOX"
Index: gen_determineargs.sh
===================================================================
--- gen_determineargs.sh	(revision 528)
+++ gen_determineargs.sh	(working copy)
@@ -209,7 +209,7 @@
 	UNIONFS_BINCACHE=`cache_replace "${UNIONFS_BINCACHE}"`
 	UNIONFS_MODULES_BINCACHE=`cache_replace "${UNIONFS_MODULES_BINCACHE}"`
 	BLKID_BINCACHE=`cache_replace "${BLKID_BINCACHE}"`
-	SUSPEND_BINCACHE=`cache_replace "${SUSPEND_BINCACHE}"`
+#	SUSPEND_BINCACHE=`cache_replace "${SUSPEND_BINCACHE}"`
   
 	DEFAULT_KERNEL_CONFIG=`arch_replace "${DEFAULT_KERNEL_CONFIG}"`
 	BUSYBOX_CONFIG=`arch_replace "${BUSYBOX_CONFIG}"`
@@ -226,7 +226,7 @@
 	UNIONFS_BINCACHE=`arch_replace "${UNIONFS_BINCACHE}"`
 	UNIONFS_MODULES_BINCACHE=`arch_replace "${UNIONFS_MODULES_BINCACHE}"`
 	BLKID_BINCACHE=`arch_replace "${BLKID_BINCACHE}"`
-	SUSPEND_BINCACHE=`arch_replace "${SUSPEND_BINCACHE}"`
+#	SUSPEND_BINCACHE=`arch_replace "${SUSPEND_BINCACHE}"`
 	
 	if [ "${CMD_BOOTSPLASH}" != '' ]
 	then
@@ -433,12 +433,12 @@
 		LVM2=0
 	fi
 
-	if isTrue "${CMD_SUSPEND}"
-	then
-		SUSPEND=1
-	else
-		SUSPEND=0
-	fi
+#	if isTrue "${CMD_SUSPEND}"
+#	then
+#		SUSPEND=1
+#	else
+#		SUSPEND=0
+#	fi
 	
 	if isTrue "${CMD_EVMS2}"
 	then
Index: genkernel.conf
===================================================================
--- genkernel.conf	(revision 528)
+++ genkernel.conf	(working copy)
@@ -132,8 +132,8 @@
 E2FSPROGS_SRCTAR="${GK_SHARE}/pkg/e2fsprogs-${E2FSPROGS_VER}.tar.gz"
 BLKID_BINCACHE="%%CACHE%%/blkid-${E2FSPROGS_VER}-%%ARCH%%.bz2"
 
-SUSPEND_VER="VERSION_SUSPEND"
-SUSPEND_DIR="suspend-${SUSPEND_VER}"
-SUSPEND_SRCTAR="${GK_SHARE}/pkg/suspend-${SUSPEND_VER}.tar.gz"
-SUSPEND_BINCACHE="%%CACHE%%/suspend-${SUSPEND_VER}-%%ARCH%%.tar.bz2"
+#SUSPEND_VER="VERSION_SUSPEND"
+#SUSPEND_DIR="suspend-${SUSPEND_VER}"
+#SUSPEND_SRCTAR="${GK_SHARE}/pkg/suspend-${SUSPEND_VER}.tar.gz"
+#SUSPEND_BINCACHE="%%CACHE%%/suspend-${SUSPEND_VER}-%%ARCH%%.tar.bz2"
 
Index: gen_initramfs.sh
===================================================================
--- gen_initramfs.sh	(revision 528)
+++ gen_initramfs.sh	(working copy)
@@ -148,23 +148,23 @@
 	rm -r "${TEMP}/initramfs-unionfs-tools-temp/"
 }
 
-append_suspend(){
-	if [ -d "${TEMP}/initramfs-suspend-temp" ];
-	then
-		rm -r "${TEMP}/initramfs-suspend-temp/"
-	fi
-	print_info 1 'SUSPEND: Adding support (compiling binaries)...'
-	compile_suspend
-	mkdir -p "${TEMP}/initramfs-suspend-temp/"
-	/bin/tar -jxpf "${SUSPEND_BINCACHE}" -C "${TEMP}/initramfs-suspend-temp" ||
-		gen_die "Could not extract suspend binary cache!"
-	mkdir -p "${TEMP}/initramfs-suspend-temp/etc"
-	cp -f /etc/suspend.conf "${TEMP}/initramfs-suspend-temp/etc" ||
-		gen_die 'Could not copy /etc/suspend.conf'
-	cd "${TEMP}/initramfs-suspend-temp/"
-	find . -print | cpio ${CPIO_ARGS} --append -F "${CPIO}"
-	rm -r "${TEMP}/initramfs-suspend-temp/"
-}
+#append_suspend(){
+#	if [ -d "${TEMP}/initramfs-suspend-temp" ];
+#	then
+#		rm -r "${TEMP}/initramfs-suspend-temp/"
+#	fi
+#	print_info 1 'SUSPEND: Adding support (compiling binaries)...'
+#	compile_suspend
+#	mkdir -p "${TEMP}/initramfs-suspend-temp/"
+#	/bin/tar -jxpf "${SUSPEND_BINCACHE}" -C "${TEMP}/initramfs-suspend-temp" ||
+#		gen_die "Could not extract suspend binary cache!"
+#	mkdir -p "${TEMP}/initramfs-suspend-temp/etc"
+#	cp -f /etc/suspend.conf "${TEMP}/initramfs-suspend-temp/etc" ||
+#		gen_die 'Could not copy /etc/suspend.conf'
+#	cd "${TEMP}/initramfs-suspend-temp/"
+#	find . -print | cpio ${CPIO_ARGS} --append -F "${CPIO}"
+#	rm -r "${TEMP}/initramfs-suspend-temp/"
+#}
 
 append_dmraid(){
 	if [ -d "${TEMP}/initramfs-dmraid-temp" ]
@@ -478,7 +478,7 @@
 #	append_data 'udev' "${UDEV}"
 	append_data 'unionfs_modules' "${UNIONFS}"
 	append_data 'unionfs_tools' "${UNIONFS}"
-	append_data 'suspend' "${SUSPEND}"
+#	append_data 'suspend' "${SUSPEND}"
 	append_data 'lvm2' "${LVM2}"
 	append_data 'dmraid' "${DMRAID}"
 	append_data 'evms2' "${EVMS2}"
Index: generic/initrd.scripts
===================================================================
--- generic/initrd.scripts	(revision 528)
+++ generic/initrd.scripts	(working copy)
@@ -847,17 +847,17 @@
 	fi
 }
 
-suspend_resume() {
-	[ -x /sbin/resume ] || return 0
-	/sbin/resume
-	local ret=$?
+#suspend_resume() {
+#	[ -x /sbin/resume ] || return 0
+#	/sbin/resume
+#	local ret=$?
 
-	if [ "${ret}" -eq 0 ]; then
-		exit 0
-	fi
+#	if [ "${ret}" -eq 0 ]; then
+#		exit 0
+#	fi
 
-	return 0
-}
+#	return 0
+#}
 
 suspend2_resume() {
 	if [ -d /proc/suspend2 ] || [ -d /sys/power/suspend2 ]; then
Index: generic/linuxrc
===================================================================
--- generic/linuxrc	(revision 528)
+++ generic/linuxrc	(working copy)
@@ -256,7 +256,7 @@
 # Run debug shell if requested
 rundebugshell
 
-suspend_resume
+#suspend_resume
 suspend2_resume
 
 if [ "${CDROOT}" -eq '1' ]
